packages/aws-cdk-lib/core/lib/errors.ts

```diff
@@ -0,0 +1,145 @@
+import { IConstruct } from 'constructs';
+import { constructInfoFromConstruct } from './helpers-internal';
+
+const CONSTRUCT_ERROR_SYMBOL = Symbol.for('@aws-cdk/core.SynthesisError');
+const VALIDATION_ERROR_SYMBOL = Symbol.for('@aws-cdk/core.ValidationError');
+
+/**
+ * Helper to check if an error is a SynthesisErrors
+ */
+export class Errors {
+  /**
+   * Test whether the given errors is a ConstructionError
+   */
+  public static isConstructError(x: any ): x is ConstructError {
+    return x !== null && typeof(x) === 'object' && CONSTRUCT_ERROR_SYMBOL in x;
+  }
+
+  /**
+   * Test whether the given error is a ValidationError
+   */
+  public static isValidationError(x: any ): x is ValidationError {
+    return Errors.isConstructError(x) && VALIDATION_ERROR_SYMBOL in x;
+  }
+}
+
+interface ConstructInfo {
+  readonly fqn: string;
+  readonly version: string;
+}
+
+/**
+ * Generic, abstract error that is thrown from the users app during app construction or synth.
+ */
+abstract class ConstructError extends Error {
+  #time: string;
```

Comment:
why `#` instead of `private`? just curious

Reply:
Confirmed they show up like this:
<img width="720" alt="image" src="https://github.com/user-attachments/assets/8f75f928-98cf-4bde-b7cb-81ec9ddbaf73">